### PR TITLE
add tasty integration notes

### PR DESCRIPTION
**Pull request by mwotton**
- [ ] Labeled the PR with `patch`, `minor`, or `major` to request a version bump when it's merged. (N/A, docs-only)
- [x] Updated the user manual in `docs/`.
- [ ] Added integration / regression tests in `tests/`. (N/A, docs-only)

### CODE REVIEW

**Comment by mwotton** [comment]
thanks to @erikryb for the patch https://github.com/MercuryTechnologies/ghciwatch/issues/317

File: docs/tasty-integration.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+main :: IO ()
+main = bubblewrap $ defaultMain $ mytestgroup
+```
+
+## tasty-discover issues
+
+If you add a new test file, the top level [tasty-discover](https://hackage.haskell.org/package/tasty-discover)
+module will not have it set up as a dependency, so will not be reloaded until you restart the `ghciwatch` process. [tasty-autocollect](https://hackage.haskell.org/package/tasty-autocollect) relies on a compiler plugin and seems to avoid 
+this problem.
```
Comment (parsonsmatt) [inline docs/tasty-integration.md:28]: This *shouldn't* be true - if `ghciwatch` sees an edit to the file that uses `tasty-discover` (or any other file that is a dependency of that module), then the GHC preprocessor plugin will run and include the newly created test file. You shouldn't need to restart ghciwatch.

File: docs/tasty-integration.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+module TestMain where
+
+import Control.Exception (SomeException, try)
+import System.IO (BufferMode (NoBuffering), hSetBuffering, stdout)
+import Test.Tasty (TestTree, defaultMain, testGroup)
+
+bubblewrap :: IO () -> IO ()
+bubblewrap io = do
+  try io :: IO (Either SomeException ())
+  hSetBuffering stdout NoBuffering
```
Comment (parsonsmatt) [inline docs/tasty-integration.md:18]: Is tasty changing the buffering mode? I'm curious if the benefit here is to set the `NoBuffering` mode, or if the benefit is just that we're flushing the buffer

**Review by 9999years — APPROVED** [review]

**Comment by 9999years** [comment]
OK, I rewrote this a little. I'm going to merge it now, let me know if you don't like any of my changes and we can sort it out. Also please try writing the top-level `Spec.hs` and see if that helps pick up new test files.

**Pull request merged**